test-clients/src/main/java/com/hedera/services/bdd/suites/reconnect/ValidateChangesAfterReconnect.java

```diff
@@ -0,0 +1,158 @@
+		final String fileInfoRegistry = "apiPermissionsReconnect";
+		final String transactionId = "specialTransactionId";
+		final String transactionFeeid = "authorizedTxn";
+		final long newFee = 159_588_904;
+
+		return customHapiSpec("validateChangesAfterReconnect")
+				.withProperties(Map.of(
+						"txn.start.offset.secs", "-5")
+				)
+				.given(
+						sleepFor(Duration.ofSeconds(25).toMillis()),
+						getAccountBalance(GENESIS)
+								.setNode("0.0.6")
+								.unavailableNode()
+				)
+				.when(
+						getAccountBalance(GENESIS)
+								.setNode("0.0.6")
+								.unavailableNode(),
+
+						fileUpdate(APP_PROPERTIES)
+								.payingWith(ADDRESS_BOOK_CONTROL)
+								.overridingProps(Map.of("minimumAutoRenewDuration", "20")),
+
+						fileCreate("effectivelyImmutable")
+								.contents("Can't touch me!"),
+						fileUpdate(API_PERMISSIONS)
+								.payingWith(ADDRESS_BOOK_CONTROL)
+								.overridingProps(Map.of("updateFile", "2-50")),
+
+						fileUpdate(EXCHANGE_RATES)
+								.contents(
+										spec -> {
+											ByteString newRates = spec
+													.ratesProvider()
+													.rateSetWith(1, 1)
+													.toByteString();
+											spec.registry().saveBytes("newRates", newRates);
+											return newRates;
+										}
+								).payingWith(MASTER),
+
+						makeFree(CryptoGetInfo),
+
+						getAccountBalance(GENESIS)
+								.setNode("0.0.6")
+								.unavailableNode()
+				)
+				.then(
+						withLiveNode("0.0.6")
+								.within(5 * 60, TimeUnit.SECONDS)
+								.loggingAvailabilityEvery(30)
+								.sleepingBetweenRetriesFor(10),
+						UtilVerbs.sleepFor(30 * 1000),
+						withLiveNode("0.0.6")
+								.within(5 * 60, TimeUnit.SECONDS)
+								.loggingAvailabilityEvery(30)
+								.sleepingBetweenRetriesFor(10),
+
+						getFileContents(API_PERMISSIONS)
+								.logged()
+								.setNode("0.0.3")
+								.payingWith(MASTER)
+								.saveToRegistry(fileInfoRegistry),
+						getFileContents(API_PERMISSIONS)
+								.logged()
+								.setNode("0.0.6")
+								.payingWith(MASTER)
+								.hasContents(fileInfoRegistry),
+
+						fileUpdate("effectivelyImmutable")
+								.setNode("0.0.6")
+								.hasPrecheck(NOT_SUPPORTED),
+
+						fileCreate("contractFile")
+								.setNode("0.0.6")
+								.payingWith(ADDRESS_BOOK_CONTROL)
+								.path(PATH_TO_VALID_BYTECODE),
+						contractCreate("testContract")
+								.bytecode("contractFile")
+								.autoRenewSecs(15)
+								.setNode("0.0.6")
+								.hasPrecheck(AUTORENEW_DURATION_NOT_IN_RANGE),
+
+						cryptoCreate("civilian")
+								.via(transactionFeeid)
+								.setNode("0.0.6"),
+						getTxnRecord(transactionFeeid)
+								.setNode("0.0.6")
+								.hasPriority(recordWith().fee(newFee)),
+
+						cryptoCreate("civilian")
+								.setNode("0.0.6"),
+						getAccountInfo("0.0.2")
+								.payingWith("civilian")
+								.nodePayment(0L)
+								.setNode("0.0.6")
+								.hasAnswerOnlyPrecheck(OK)
+				);
```

Comment:
Before ending, we should restore all of the protected system files back to their original contents.

Reply:
@qnswirlds once the test ends, the machines are terminated. What would be the point of restoring those files?

Reply:
When we create a test, we don't limit ourselves to use such test only on a "testing" environment. The test could be used in prod environment like our testnet or mainnet. In this case, clean up is definitely required. Besides, cleaning up after performing any test is a better practice to follow in general.

Reply:
Fixed